### ISSUE TITLE
Question: is OpenCV error in opening MP4 files to extract dropped frames from a store version-related? 

### ISSUE THREAD

**Issue by vincejstraub** [loopbio/imgstore#22]
#### Description

My aim is to extract the number of frames dropped by the Loopbio but when trying to read a store (`VideoImgStore`) using the `imgstore` library in a Jupyter notebook, I receive an error indicating that OpenCV is unable to open MP4 files. Can the error I am receiving be resolved by installing a specific version of OpenCV for python? Is there a way to extract the amount of dropped frames using command line tools?

#### To Reproduce
Requirements:

```
imgstore: 0.2.9
opencv-python: 4.5.2
jupyter core: 4.6.3
jupyter-notebook: 6.0.3
jupyter lab 3.0.14
```

Run the following argument in a Jupyter notebook: 

```
from imgstore import new_for_filename

store = new_for_filename('20200701_094933/metadata.yaml')
```
where `20200701_094933` is a directory containing NPZ and MP4 chunks, and a YAML metadata file generated by a Loopbio video recording system. Hopefully you should also get the following error message:

```

---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-50-b5542c48ff88> in <module>
      1 from imgstore import new_for_filename
      2 
----> 3 store = new_for_filename('20200701_094933/metadata.yaml')

~/anaconda3/lib/python3.7/site-packages/imgstore/stores.py in new_for_filename(path, **kwargs)
   1193         raise ValueError('store class %s not supported' % clsname)
   1194 
-> 1195     return cls(**kwargs)
   1196 
   1197 

~/anaconda3/lib/python3.7/site-packages/imgstore/stores.py in __init__(self, **kwargs)
    976             kwargs['encoding'] = kwargs.pop('encoding', None)
    977 
--> 978         _ImgStore.__init__(self, **kwargs)
    979 
    980         self._supports_seeking = seek

~/anaconda3/lib/python3.7/site-packages/imgstore/stores.py in __init__(self, basedir, mode, imgshape, imgdtype, chunksize, metadata, encoding, write_encode_encoding)
    138 
    139             # reset to the start of the file and load the first chunk
--> 140             self._load_chunk(0)
    141             assert self._chunk_current_frame_idx == -1
    142             assert self._chunk_n == 0

~/anaconda3/lib/python3.7/site-packages/imgstore/stores.py in _load_chunk(self, n, _force)
   1066 
   1067             if not self._cap.isOpened():
-> 1068                 raise Exception("OpenCV unable to open %s" % fn)
   1069 
   1070             self._load_chunk_metadata(os.path.join(self._basedir, '%06d' % n))

Exception: OpenCV unable to open 20200701_094933/000000.mp4

```

#### Checklist
- [ ] After running run the command `imgstore-test` and receiving the error message `The opencv backend does not actually have write support`, I tried installing different versions of OpenCV that should support writing h264 encoded videos but still received the above exception. Specifically, I have tried using `3.4.2.17`; `4.5.2.52`; `3.4.14.51`.
- [ ] I have tried install `pip install opencv-contrib-python`, as suggested on [Stack Overflow](https://stackoverflow.com/questions/49299328/opencv-cannot-open-mp4-file)
- [ ] I have tried solutions to the exception `OpenCV unable to open %s" % fn)` offered on [Stack Overflow ](https://stackoverflow.com/questions/59371075/opencv-error-cant-open-camera-through-video-capture)to no avail.

**Comment by nzjrs** [comment]
1. Is the 000000.mp4 video actually readable and non-corrupt? Can it be opened in VLC for example?

2. Can the current virtual environment / jupyterlab install successfully open the sample imgstore In https://github.com/loopbio/imgstore/tree/master/imgstore/tests/data/store_mp4

**Comment by vincejstraub** [comment]
1. The 000000.mp4 video is readable; I was able to open it with VLC and other media players.
2. The jupyterlab notebook which I am using in a virtual environment was not able to successfully open the sample imgstore.
 
After inspecting the sample imgstore, I was, however, able to find out that the problem was due to a dependency/opencv version issue; I needed to install [opencv-contrib-python](https://pypi.org/project/opencv-contrib-python/) (i.e. FFMPEG) for opencv-python to be able to open my store (class: VideoImgStoreFFMPEG). 

Thank you for the quick reply and for pointing out the sample imgstore!

**Comment by vincejstraub** [comment]
1. The 000000.mp4 video is readable; I was able to open it with VLC and other media players.
2. The jupyterlab notebook which I am using in a virtual environment was not able to successfully open the sample imgstore.
 
After inspecting the sample imgstore, I was, however, able to find out that the problem was due to a dependency/opencv version issue; I needed to install [opencv-contrib-python](https://pypi.org/project/opencv-contrib-python/) (i.e. FFMPEG) for opencv-python to be able to open my store (class: VideoImgStoreFFMPEG). 

Thank you for the quick reply and for pointing out the sample imgstore!
